src/InsertCapableInterface.php

```diff
@@ -19,7 +16,11 @@
      *
      * @since [*next-version*]
      *
-     * @param array[]|ArrayAccess|stdClass|ContainerInterface|Traversable $records A list of record data containers.
+     * @param array|Traversable $records A list of records data containers. Accepted container types are:
```

Comment:
This solution prevents static code analysis by existing tools.

Reply:
Then please suggest an alternative.

Reply:
Actually, why is the above a problem? What about the above exactly prevents static code analysis? And does the code _really_ have to change to accommodate such tools?